packages/core/src/containers/modules/artifactBuilder.ts

```diff
@@ -26,24 +22,36 @@ export const artifactBuilderModule = (options: IArtifactBuilderOptions) =>
       .inSingletonScope();
 
     // PersistentStore
-    bind<PersistentStore>(TYPES.PersistentStore)
-      .to(LocalFilePersistentStore)
-      .inSingletonScope()
-      .whenTargetNamed(PersistentStoreType.LocalFile);
-
     bind<interfaces.AutoNamedFactory<PersistentStore>>(
       TYPES.Factory_PersistentStore
-    ).toAutoNamedFactory<PersistentStore>(TYPES.PersistentStore);
+    ).toAutoNamedFactory<PersistentStore>(TYPES.Extension_PersistentStore);
+    bind<PersistentStore>(TYPES.PersistentStore)
+      .toDynamicValue((context) => {
+        const factory = context.container.get<
+          interfaces.AutoNamedFactory<PersistentStore>
+        >(TYPES.Factory_PersistentStore);
+        const options = context.container.get<ArtifactBuilderOptions>(
+          TYPES.ArtifactBuilderOptions
+        );
+        return factory(options.provider);
```

Comment:
> Here also have a question, for the `AutoNameFactory`, the factory needs a target name to locate the mapping class object, but here I do not see you use the `whenTargetNamed`, how do you let the container name factory could locate, and here you target the name?

Named bindings are done by extension loader. `extension-loader/extensionLoader.ts`
```ts
 extensionBinding.when((request) => {
        // If request contains named tag, i.e. @named or getNamed(), we check the extensionId, otherwise we fulfill the request.
        // It makes both @named tag and @multiInject work at same time.
        const namedTag = request.target.getNamedTag();
        if (namedTag) return namedTag.value === extensionId;
        return true;
});
```

Reply:
Ohoh! Got it, thanks for @oscar60310 solving my question!